### ISSUE TITLE
Details about the parallel execution

### ISSUE THREAD

**Issue by pedgarcia** [yandex/mocha-parallel-tests#106]
This is not really an issue, but I couldn't find the information elsewhere, so I hope it is ok to ask here...

I understand that the level of parallelism is between specs (or files). Is that correct?

For example:

specs/suite_a/file_spec1a.js
specs/suite_a/file_spec2a.js
..
specs/suite_b/file_spec1b.js
specs/suite_b/file_spec2b.js

If I run pointing to specs/* I could see that **suite_a/file_spec1a.js** and **suite_a/file_spec2a.js** will run in parallel.

How about files in different suites: will **suite_a/file_spec1a.js** and **suite_b/file_spec??.js** run also in parallel?

Thanks in advance!

Paulo

**Comment by 1999** [comment]
Hey @pedgarcia. Currently mocha-parallel-tests runs all suites in parallel (i.e. even all `describe`s inside one file will run in parallel). All test cases (`it`) run one after another. This is probably not the ideal behaviour especially because some developers write test files like this:

```javascript
describe('module', () => {
  describe('submodule A', () => {
    describe('feature B', () => { ... })
  })
})
```

But this is current behaviour and it seems like it suits the majority of the tasks.

**Comment by pedgarcia** [comment]
Hm, ok. Now I understand! The current behavior is actually ideal for me as well. I might have some nested describes but this can probably be easily changed. 

Thanks a lot. This is a great and useful project!

**Comment by 1999** [comment]
@pedgarcia thanks for nice words mate :)

**Comment by pedgarcia** [comment]
@1999 I hope you still see this since it is closed... What happens when you have nested describes in terms of parallelization?

Imagine the following scenario:
```
describe('module', () => {
  describe('submodule A', () => {
    describe('feature B', () => { ... })
         it(A..)
         it(B..)
  });
  it(C...)
  it(D...)
})
```
Will A B C and D run in parallel, or A and B will, then C and D will, but only after A and B?

If you have better examples, please point me at them

Thanks


Paulo

**Comment by 1999** [comment]
@pedgarcia I created a separate issue for this, let's discuss it there: #110
